glib/src/gobject/auto/type_plugin.rs

```diff
@@ -0,0 +1,83 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use crate::{prelude::*, translate::*, InterfaceInfo, TypeInfo, TypeValueTable};
+use std::fmt;
+
+crate::wrapper! {
+    #[doc(alias = "GTypePlugin")]
+    pub struct TypePlugin(Interface<gobject_ffi::GTypePlugin, gobject_ffi::GTypePluginClass>);
+
+    match fn {
+        type_ => || gobject_ffi::g_type_plugin_get_type(),
+    }
+}
+
+impl TypePlugin {
+    pub const NONE: Option<&'static TypePlugin> = None;
+}
+
+mod sealed {
+    pub trait Sealed {}
+    impl<T: super::IsA<super::TypePlugin>> Sealed for T {}
+}
+
+pub trait TypePluginExt: IsA<TypePlugin> + sealed::Sealed + 'static {
+    #[doc(alias = "g_type_plugin_complete_interface_info")]
+    fn complete_interface_info(
```

Comment:
What I mean is that this PR adds quite a bit of public API that should probably never be called by user code but only by the macro, so maybe it can be moved directly into the macro instead and call the FFI API there.

`TypePlugin` / TypeModule` and their ext traits with the (useful) public API should still exists of course.